catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,310 @@
+        entry.setChildren(children);
+        return entry;
+    }
+
+    /**
+     * List the files. Creates entries for each child. Return should be set on parent.
+     *
+     * @param file  The file to list files for
+     * @param entry the parent entry
+     * @return The child files
+     */
+    private DavEntry[] doListFiles(String file, DavEntry entry) {
+        final DavResource[] files = listFiles(file);
+        final DavEntry[] children =
+                files.length > 0 ? new DavEntry[files.length] : DavEntry.getEmptyEntries();
+        for (int i = 0; i < files.length; i++) {
+            children[i] = createFileEntry(entry, files[i]);
+        }
+        return children;
+    }
+
+    /**
+     * Fire directory/file created events to the registered listeners.
+     *
+     * @param entry The file entry
+     */
+    private void doCreate(final DavEntry entry) {
+        for (final EntryAlterationListener listener : listeners) {
+            if (entry.isDirectory()) {
+                listener.onDirectoryCreate(entry);
+            } else {
+                listener.onFileCreate(entry);
+            }
+        }
+        final DavEntry[] children = entry.getChildren();
+        for (final DavEntry aChildren : children) {
+            doCreate(aChildren);
+        }
+    }
+
+    /**
+     * Fire directory/file change events to the registered listeners.
+     *
+     * @param entry The previous file system entry
+     * @param file  The current file
+     */
+    private void doMatch(final DavEntry entry, final DavResource file) {
+        if (entry.refresh(file)) {
+            for (final EntryAlterationListener listener : listeners) {
+                if (entry.isDirectory()) {
+                    listener.onDirectoryChange(entry);
+                } else {
+                    listener.onFileChange(entry);
+                }
+            }
+        }
+    }
+
+    /**
+     * Fire directory/file delete events to the registered listeners.
+     *
+     * @param entry The file entry
+     */
+    private void doDelete(final DavEntry entry) {
+        for (final EntryAlterationListener listener : listeners) {
+            if (entry.isDirectory()) {
+                listener.onDirectoryDelete(entry);
+            } else {
+                listener.onFileDelete(entry);
+            }
+        }
+    }
+
+    /**
+     * List the contents of a directory
+     *
+     * @param file The file to list the contents of
+     * @return the directory contents or a zero length array if
+     * the empty or the file is not a directory
+     */
+    private DavResource[] listFiles(final String file) {
+        DavResource[] children = null;
+        try {
+            List<DavResource> list = sardine.list(file);
+            // the returned list includes the parent
+            if (list.size() > 1 && list.get(0)
+                    .isDirectory()) {
+                children = list.subList(1, list.size())
```

Comment:
Almost. `sort` returns void so it can't be chained like that. Definitely don't need the extra conversions though. 😃